pkg/cli/cvd.go

```diff
@@ -226,76 +228,31 @@ func listCVDs(c *cobra.Command, flags *ListCVDsFlags, opts *subCommandOpts) erro
 			hosts = append(hosts, host.Name)
 		}
 	}
-	var wg sync.WaitGroup
-	var mu sync.Mutex
-	var merr error
-	sortedWriter := newSortedWriter(c.OutOrStdout())
-	for i, host := range hosts {
-		wg.Add(1)
-		go func(index int, name string) {
-			defer wg.Done()
-			hostCVDs, err := service.ListCVDs(name)
-			mu.Lock()
-			defer mu.Unlock()
+	var chans []chan result[[]*hoapi.CVD]
+	for _, host := range hosts {
+		ch := make(chan result[[]*hoapi.CVD])
+		chans = append(chans, ch)
+		go func(name string, ch chan<- result[[]*hoapi.CVD]) {
+			cvds, err := service.ListCVDs(name)
 			if err != nil {
-				merr = multierror.Append(merr,
-					fmt.Errorf("lists cvds for host %q failed: %w", name, err))
+				ch <- result[[]*hoapi.CVD]{Error: err}
 				return
 			}
-			out := &bytes.Buffer{}
-			printCVDs(out, buildBaseURL(flags.CVDRemoteFlags), name, hostCVDs)
-			b, _ := io.ReadAll(out)
-			sortedWriter.Push(index, string(b))
-		}(i, host)
-	}
-	wg.Wait()
-	sortedWriter.Flush()
-	return merr
-}
-
-// Writes payloads in order.
-type sortedWriter struct {
-	w           io.Writer
-	payloadsMap map[int]string
-	currIndex   int
-	mu          sync.Mutex
-}
+			ch <- result[[]*hoapi.CVD]{Result: cvds}
```

Comment:
nit: `ch <- result[[]*hoapi.CVD]{cvds, err}` would avoid the `if`

Reply:
agree, done